src/App.jsx

```diff
@@ -14,33 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+// export const handleLengthSort = () => {
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+// };
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+const SORT_ALPHABET = 'alphabetic';
+const SORT_LENGTH = 'length';
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const sortGoods = (goods, params) => {
+  const { sortBy, isReversed } = params;
+  const sortedGoods = [...goods];
+
+  if (sortBy.length) {
+    sortedGoods.sort((good1, good2) => {
+      if (sortBy === SORT_ALPHABET) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortBy === SORT_LENGTH) {
+        return good1.length - good2.length;
+      }
+
+      throw new Error('There is no such type of sort');
+    });
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sortedGoods = sortGoods(goodsFromServer, { sortBy, isReversed });
+
+  const resetAll = () => {
+    setSortBy('');
+    setReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortBy !== SORT_ALPHABET,
+          })}
+          onClick={() => setSortBy(SORT_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortBy !== SORT_LENGTH,
+          })}
+          onClick={() => setSortBy(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {sortBy || isReversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => resetAll()}
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
You don't need to use the ternary operator if you return nothing in the second case, use && instead